### PR TITLE
#151435474 Add tags as a required field when creating a new contribution

### PR DESCRIPTION
**Pull request by andela-ieyo**
#### What does this PR do? ####

       -  Refactor the POST view function to include `tags` a required field
       -  Refactor validateRequest helper function to validate that tags is present in the request 
           payload. 
       -  Create endpoint for GET all contribution_types.
       -  write the view function to handle GET all contribution_types.
       -  update unit tests to reflect changes.

#### How should this be manually tested? ####

       At the root of the project directory run the following command:
      $ DB_TYPE=sqlite pytest -v OR $ DB_TYPE=sqlite pytest

#### What are the relevant PT stories? ####

       [151435474] [151243725]

### CODE REVIEW

**Comment by andela-milesanmi** [comment]
@andela-ieyo Nice work. Few things though; you have 2 commits; has the `url` format been confirmed either `contribution/types` or `contributions/types`.

Otherwise, LGTM.

File: api/views/contributions.py [hunk 1]
```diff
@@ -105,7 +105,13 @@ def post(self):
         user_id = g.current_user.user_id
         user_name = g.current_user.name
         payload = request.get_json()
-        contribution_type = ContributionType().query.get(payload["type"])
+        if 'tags' not in payload:
```
Comment (LarryWachira) [inline api/views/contributions.py]: Develop has a validate request decorator in `helper.py` done by @andela-ojoloko. Consider using that to keep your code lean.

File: api/views/contributions.py [hunk 2]
```diff
@@ -105,7 +105,13 @@ def post(self):
...
+            return {
+                "status": "fail",
+                "data": {"message":
+                        "tags is required"}
+            }, 400
+        contribution_type = ContributionType.query.get(payload["type"])
```
Comment (LarryWachira) [inline api/views/contributions.py]: All Models now have a `get()` method that wraps `query.get()`.

File: api/views/contributions.py [hunk 3]
```diff
@@ -609,3 +614,18 @@ def get(self, notif_id=None):
             ))
         response.status_code = 200
         return response
+
+
+class ContributionTypeResource(Resource):
+
+    @token_required
+    def get(self):
+        contribution_types =  [contrib_type.serialize() for contrib_type in ContributionType.query.all()]
```
Comment (LarryWachira) [inline api/views/contributions.py]: Same here, use `fetch_all()` instead of `query.all()`

**Comment by andela-ieyo** [comment]
@mbilesanmi Yes...the endpoint is now `/api/v1/contribution_types`....approved by Njirap and Kesiena

File: main.py [hunk 4]
```diff
@@ -141,6 +141,12 @@ def index():
                      '/api/v1/skills/',
                      endpoint='skills')
 
+    # contributionType endpoint
+    api.add_resource(ContributionTypeResource,
+                    '/api/v1/contribution/types',
```
Comment (LarryWachira) [inline main.py]: You also need to update the URL as per the feedback from Njira and Kes.

Comment (andela-ieyo) [reply main.py]: @larry the helper.js doesn't cover for `tags` not in payload, it only covers for the data type of `tags` when it is present in the payload...it is either i update that or keep the present approach.

Comment (LarryWachira) [reply main.py]: Ok, got it!

**Comment by LarryWachira** [comment]
LGTM

**Pull request merged**